src/couch_replicator_scheduler.erl

```diff
@@ -404,3 +410,97 @@ update_history(Job, Type, When, State) ->
     History0 = [{Type, When} | Job#job.history],
     History1 = lists:sublist(History0, State#state.max_history),
     Job#job{history = History1}.
+
+
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+
+
+backoff_msec_test() ->
+    BaseMicros = ?BACKOFF_INTERVAL_MICROS,
+    ?assertEqual(backoff_msec(1), BaseMicros),
+    ?assertEqual(backoff_msec(2), 2 * BaseMicros),
+    ?assertEqual(backoff_msec(3), 4 * BaseMicros),
+    MaxBackoff = (1 bsl ?MAX_BACKOFF_EXPONENT) * BaseMicros,
```

Comment:
final note: we should split these out into more tests, each assert here is a separate test. should we break something, we'd like to know which of these asserts fail, rather than only the one which fails first of potentially many failures.